File: ex_clt/Th_effects.m
function NM_th = Th_effects(Cst,delta,alpha_r,z)
    % Th_effects returns N^Th and M^Th
    %
    % Arguments:
    %   alpha_r : cell array of size n with alpha vectors in ply CS
    %   z   : Array with size n+1 with location of ply edges
    %
    % Returns:
    %   NM_th : N^Th and M^th vector (6 by 1)
    N = zeros(3,1);
    M = zeros(3,1);
    if length(z) - 1 == length(alpha_r)
        for i=1:length(alpha_r)
            N = N + delta*Cst{i}*alpha_r{i}*(z(i+1)-z(i));
            M = M + delta*Cst{i}*alpha_r{i}/2*(z(i+1)^2-z(i)^2);
        end
    end
    NM_th = [N; M];
end

File: Python/th_effects.py
# -*- coding: utf-8 -*-
"""
Created on Wed Nov 27 15:25:34 2024

@author: Timme
"""
import numpy as np
def th_effects(Cst,delta,alpha_r,z):
    """
    

    Parameters
    ----------
    Cst : array
        Array of size n of rotated stifness matrices C*s.
    delta : scalar
        Temperature difference.
    alpha_r : array
        array of vectors of rotated thermal coefficients.
    z : array
        Array of size n+1 with location of ply edges.

    Returns
    -------
    NM_th : vector
        N^th and M^th vector (6 by 1).

    """
    #M = np.zeros(3)
    #N = np.zeros(3)
    lst1 = []
    if len(z) -1 == len(alpha_r):
       # for i in alpha_r:
        #    Ni = Cst[i]*alpha_r[i]*(z[i+1]-z[i])
         #   lst1.append(Ni)
        Nx = [Cst[i]*alpha_r[i]*(z[i+1]-z[i]) for i in alpha_r] # calculation of each Nth force of each layer
        Mx = [Cst[i]*alpha_r[i]*(z[i+1]**2-z[i]**2) for i in alpha_r] # calculation of the Mth force of each layer
        Nx = np.array(Nx) # convert to numpy array for faster computation
        Mx = np.array(Mx)
        N = delta*np.sum(Nx) # sum the forces, multiplied by the temperature difference
        M = delta/2*np.sum(Mx)

        #N = [N+]